File: engine/engine-model/src/main/scala/tools/packet/showitemgaininchat/ShowOwnBerserk.scala
package tools.packet.showitemgaininchat

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class ShowOwnBerserk(private var _skillLevel: Int, private var _berserk: Boolean) extends PacketInput {
  def skillLevel: Int = _skillLevel

  def berserk: Boolean = _berserk

  override def opcode(): SendOpcode = SendOpcode.SHOW_ITEM_GAIN_INCHAT
}

File: engine/engine-model/src/main/java/tools/packet/showitemgaininchat/ShowOwnBerserk.java
package tools.packet.showitemgaininchat;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record ShowOwnBerserk(Integer skillLevel, Boolean berserk) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.SHOW_ITEM_GAIN_IN_CHAT;
   }
}